utils.py

```diff
@@ -809,6 +809,39 @@ def polylcm(a, b):
         lcm_roots = a_roots + b_roots
         return numpy.poly1d(lcm_roots, True)
 
+def multi_polylcm(P):
+    roots_list = [i.r.tolist() for i in P]
+    roots_by_mult = []
+    lcm_roots_by_mult = []
+    for roots in roots_list:
+        root_builder = []
+        for root in roots:
+            repeated = False
+            for i in range(len(root_builder)):
```

Comment:
`for x in range` is a bit of an antipattern in Python. Especially this form `range(len(`. Check out https://nedbatchelder.com/text/iter.html